reference_test.go

```diff
@@ -41,48 +42,62 @@ func isSubunitInstalled() bool {
-	{"successful-test", "success"},
-	{"unexpected-successful-test", "uxsuccess"},
-	{"skipped-test", "skip"},
-	{"failed-test", "fail"},
-	{"expected-failed-test", "xfail"},
+	{TestID: "existing-test", Status: "exists"},
+	{TestID: "progressing-test", Status: "inprogress"},
+	{TestID: "successful-test", Status: "success"},
+	{TestID: "unexpected-successful-test", Status: "uxsuccess"},
+	{TestID: "skipped-test", Status: "skip"},
+	{TestID: "failed-test", Status: "fail"},
+	{TestID: "expected-failed-test", Status: "xfail"},
 
 	// Different test id lengths.
-	{"test-id (1 byte)", "exists"},
-	{"test-id-with-63-chars (1 byte____)" + strings.Repeat("_", 63-34), "exists"},
-	{"test-id-with-64-chars (2 bytes___)" + strings.Repeat("_", 64-34), "exists"},
-	{"test-id-with-16383-chars (2 bytes)" + strings.Repeat("_", 16383-34), "exists"},
-	{"test-id-with-16384-chars (3 bytes)" + strings.Repeat("_", 16384-34), "exists"},
+	{TestID: "test-id (1 byte)", Status: "exists"},
+	{TestID: "test-id-with-63-chars (1 byte____)" + strings.Repeat("_", 63-34), Status: "exists"},
+	{TestID: "test-id-with-64-chars (2 bytes___)" + strings.Repeat("_", 64-34), Status: "exists"},
+	{TestID: "test-id-with-16383-chars (2 bytes)" + strings.Repeat("_", 16383-34), Status: "exists"},
+	{TestID: "test-id-with-16384-chars (3 bytes)" + strings.Repeat("_", 16384-34), Status: "exists"},
 	// We can't test IDs with more length bytes through the command line.
+
+	// Test with timestamp.
+	// Round to microseconds because python's datetime does not accept nanoseconds.
+	{TestID: "test-with-timestamp", Status: "success", Timestamp: time.Now().Round(time.Microsecond)},
 }
 
 func (s *SubunitReferenceSuite) TestReference(c *check.C) {
-	for _, t := range referencetests {
+	for _, e := range referencetests {
 		var goOutput bytes.Buffer
 		stream := &subunit.StreamResultToBytes{Output: &goOutput}
-		err := stream.Status(t.id, t.status)
-		c.Check(err, check.IsNil, check.Commentf("Error running the go version of subunit"))
+		err := stream.Status(e)
+		c.Check(err, check.IsNil, check.Commentf("Error running the go version of subunit: %s", err))
 
 		cmd := exec.Command("python", "-c", fmt.Sprintf(
 			// FIXME the runnable flag must be a parameter. --elopio - 2015-08-31
-			"import subunit; import sys; subunit.StreamResultToBytes(sys.stdout).status(test_id=%q, test_status=%q, runnable=False)",
-			t.id, t.status))
+			"import datetime; import subunit; import sys; from subunit import iso8601; "+
+				"subunit.StreamResultToBytes(sys.stdout).status(%s)",
+			makePythonArgs(e)))
 		pythonOutput, err := cmd.Output()
-		c.Check(err, check.IsNil, check.Commentf("Error runninng the python version of subunit"))
+		c.Check(err, check.IsNil,
+			check.Commentf("Error runninng the python version of subunit: %s", err))
 
-		c.Check(goOutput.Bytes(), check.DeepEquals, pythonOutput)
+		c.Check(goOutput.Bytes(), check.DeepEquals, pythonOutput,
+			check.Commentf("Wrong stream for event %v", e))
```

Comment:
I think that's because with Now() you get a different timezone. I pushed a cast to UTC.